widget/tabcontainer.go

```diff
@@ -411,13 +414,78 @@ var _ desktop.Hoverable = (*tabButton)(nil)
 type tabButton struct {
 	BaseWidget
 	hovered      bool
+	focused      bool
+	pressed      bool
 	Icon         fyne.Resource
 	IconPosition buttonIconPosition
 	OnTap        func()
 	Style        ButtonStyle
 	Text         string
 }
 
+// FocusGained is called when the Entry has been given focus.
+func (b *TabContainer) FocusGained() {
+	b.focused = true
+	b.Refresh()
```

Comment:
Refreshing a whole container is expensive (as it invalidates all children).
Here you are not actually using the focused value in the container but in it's tabs.
If the tabs tracked focus and/or the need to refresh it would be much more performant

Reply:
Yes, it is actualy only the buttons that need to be refreshed. But since the TabContainer have no refrences to the buttons, it is difficult to refresh only the buttons. I will leave this for later.